### PR TITLE
libcu++: convert the remaining fpemu unit tests to lit

### PR DESCRIPTION
**Pull request by akolesov-nvidia**
The Catch2 .cu tests for comparison, division, float conversion, integer conversion and sqrt become lit .pass.cpp tests next to the ones already converted, so the whole fpemu suite runs under one harness.

Adds unary.pass.cpp, which covers the increment, decrement, negation and compound assignment operators that had no test at all.

## Description

closes 

## Checklist

- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

File: libcudacxx/test/libcudacxx/cuda/fp/units/fpemu/cmp.pass.cpp [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+  OP_EQ = 0,
+  OP_NE,
+  OP_LT,
+  OP_LE,
+  OP_GT,
+  OP_GE,
+  OP_COUNT
+};
+
+_CCCL_HOST_DEVICE double from_bits(uint64_t b)
```
Comment (miscco) [inline libcudacxx/test/libcudacxx/cuda/fp/units/fpemu/cmp.pass.cpp]: Nitpick: In tests we historically use `TEST_HOST_DEVICE_FUNC`

Its meant to be different from the `_CCCL_` macro for some testing edge cases

File: libcudacxx/test/libcudacxx/cuda/fp/units/fpemu/cmp.pass.cpp [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+#include <cuda/fpemu>
+#include <cuda/std/bit>
+#include <cuda/std/cassert>
+#include <cuda/std/cstdint>
+#include <cuda/std/random>
+
+#include "test_macros.h"
+
+using namespace cuda::experimental; // FP SDK lives in cuda::experimental (later cuda::)
```
Comment (miscco) [inline libcudacxx/test/libcudacxx/cuda/fp/units/fpemu/cmp.pass.cpp]: Important: `using namespace` is really dangerous practice and we try hard to never use it.

If this is too long please use
```cpp
namespace cudax = cuda::experimental;
```

And use that in the test

**Review by miscco — CHANGES_REQUESTED** [review]

File: libcudacxx/test/libcudacxx/cuda/fp/units/fpemu/cmp.pass.cpp (hunk 2 above)
Comment (akolesov-nvidia) [reply libcudacxx/test/libcudacxx/cuda/fp/units/fpemu/cmp.pass.cpp]: Done

File: libcudacxx/test/libcudacxx/cuda/fp/units/fpemu/cmp.pass.cpp (hunk 1 above)
Comment (akolesov-nvidia) [reply libcudacxx/test/libcudacxx/cuda/fp/units/fpemu/cmp.pass.cpp]: Ok, renamed them all

**Review by davebayer — APPROVED** [review]

File: libcudacxx/test/libcudacxx/cuda/fp/units/fpemu/pi.pass.cpp [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+  check_pi<cudax::fp64emu_unpacked_high>(one, iters, tol);
+  check_pi<cudax::fp64emu_unpacked_mid>(one, iters, tol);
+  check_pi<cudax::fp64emu_unpacked_low>(one, iters, tol_low);
+}
+
+int main(int, char**)
+{
+  test();
+
+  return 0;
```
[the remark this answers is not in the document]
Comment (davebayer) [reply libcudacxx/test/libcudacxx/cuda/fp/units/fpemu/pi.pass.cpp:155]: lit-style tests run main on both host and device

**Review by miscco — APPROVED** [review]
